### PR TITLE
Addresses issues with zero order being un-sortable (and pathogenic)

### PR DESCRIPTION
**Pull request by discorick**
Fix https://github.com/huboard/huboard-web/issues/159

### CODE REVIEW

**Comment by dahlbyk** [comment]
Paired with @discorick on this. Had started with a client-side solution, but we can avoid that complexity by always forcing `order` and `milestone_order` to have positive, deterministic values. Two side effects from this solution:

1. `0`-order issues now sort by issue `id` instead of by repository name and issue number. Since this sort order could not have been deliberate, this seems acceptable.
2. While the new "default" sort order is intentionally very small, there is a chance that existing non-`0` issues will have a smaller `order` or `milestone_order` value, effectively injecting the `0` issues between otherwise prioritized issues. This could be mitigated by picking an even smaller value for `zero_adjustment`, but I'm content with the current value.

**Comment by rauhryan** [comment]
Ok, seems legit. 

:shipit: ?

**Comment by dahlbyk** [comment]
@discorick while you're testing this, can you check how we handle `order`/`milestone_order` values that are not numeric? I would expect them to get the same treatment as non-positive numeric values.

**Comment by discorick** [comment]
@dahlbyk hacking the order to a string throws a 422, which is a legitimate response - it prevents the board from loading though.

**Comment by discorick** [comment]
@dahlbyk review 06e2ce2 ?

File: lib/bridge/github/issues.rb [hunk 1]
```diff
@@ -232,7 +232,19 @@ def embed_data(data = nil)
           begin
             data = MultiJson.load(match[1])
             data["order"] = self['number'] unless data["order"]
+
+            zero_adjustment = 0.1e-20
+            if data["order"] <= 0 || data["order"].is_a?(String)
```
Comment (dahlbyk) [inline lib/bridge/github/issues.rb]: Seems safer to check if it's a number first, then compare to 0?

**Comment by dahlbyk** [comment]
:shipit:

**Pull request merged**